src/coreComponents/mainInterface/ProblemManager.cpp

```diff
@@ -580,6 +580,9 @@ void ProblemManager::generateMesh()
       edgeManager.setGeometricalRelations( cellBlockManager );
       faceManager.setGeometricalRelations( cellBlockManager, nodeManager );
 
+      nodeManager.buildRegionMaps( elemManager );
+      faceManager.buildRegionMaps( elemManager );
```

Comment:
Why do you have to move those 2 lines up?

For the `nodeManager` (and probably for the other managers in a similar way), I assume that the data (more precisely `NodeManager::m_toElements.m_toElementIndex`) that is defined in `NodeManager::setGeometricalRelations` are not valid anymore because they get overwritten in `NodeManager::buildRegionMaps`.

The problem with this PR is that it blurs the clear computational boundaries and responsibilities.
- `NodeManager::setGeometricalRelations` is not alone responsible for setting up the geometrical relations anymore.
- `NodeManager::buildRegionMaps` is not _only_ building the region maps anymore since it also _rewrites geometrical relations_. This is why we end up inserting business (reservoir, mechanics) info in the middle of the geometrical process.

I think this is what concerns me the most about the PR.

Reply:
I didn't move them above `setGeometricalRelations()` calls though, just above some other unrelated stuff like `buildSets()`, on which we have no dependency.

I do see your point, but I disagree that `buildRegionMaps` is rewriting geometrical relations, because it's taking its source data purely from element-to-(node/face) maps that were built from `CellBlockManager`, just not reusing the redundant (node/face)-to-element maps. In a way, this is better for ensuring consistency between direct/reverse mappings, rather than relying on all `CellBlockManagerABC` implementations to do that.